File: 03-secondary-indexes/src/queryRoles.js
const AWS = require("aws-sdk");
const DynamoDB = new AWS.DynamoDB({ region: "us-east-1" });

DynamoDB.query(
  {
    TableName: "MovieRoles",
    IndexName: "MovieActorIndex",
    KeyConditionExpression: "#movie= :movie",
    ExpressionAttributeNames: {
      "#movie": "Movie",
    },
    ExpressionAttributeValues: {
      ":movie": { S: "Toy Story" },
    },
  },
  function (err, data) {
    if (err) console.log(`Error performing query": ${err}`);
    else
      console.log(
        `Found ${data.Items.length} roles!\n${JSON.stringify(data.Items)}`
      );
  }
);

File: 03-secondary-indexes/python/query_roles.py
import json

import boto3

client = boto3.client("dynamodb", region_name="us-east-1")

try:
    response = client.query(
        TableName="MovieRoles",
        IndexName="GenreYearIndex",
        KeyConditionExpression="#genre= :genre",
        ExpressionAttributeNames={"#genre": "Genre"},
        ExpressionAttributeValues={":genre": {"S": "Drama"}},
    )
    print(
        f"Found {len(response['Items'])} roles!\n{json.dumps(response['Items'], indent=2)}"
    )
except Exception as err:
    print(f"Error fetching roles: {err}")
